Curation: guidelines for reviewers and approvers (#185)
codewars/docs: content/authoring/index.md

next: /authoring/kata/
---

# Authoring of Codewars Content

## Overview

> All content available on Codewars is created by members of the Codewars community. It's ideas of Codewars users what gets turned into kata, and only the fact that users are familiar with plenty of programming languages allows for a large amount of available translations.

Reviewer: ```suggestion
All content available on Codewars is created by members of the Codewars community. It's ideas of Codewars users that get turned into kata, and only the fact that users are familiar with plenty of programming languages allows for a large number of available translations.
```

> This section of documentation is meant mostly for authors and translators, who would like to introduce new challenges and make them available for other Codewars users. Creting a kata or translating it, even an easy one, is a difficult task. It requires much wider set of skills compared to the one required for just solving a kata. Unlike problem solvers, authors need to master skills related to software developemnt in general, as well as to design of the problems, writing good test suites, documenting the task and its requirements, and communicating with reviewers. 

Reviewer: ```suggestion
This section of the documentation is meant mostly for authors and translators, who would like to introduce new challenges and make them available for other Codewars users. Creating a kata or translating it, even an easy one, is a difficult task. It requires a much wider set of skills compared to the one required for just solving a kata. Unlike problem solvers, authors need to master skills related to software development in general, as well as to design of the problems, writing good test suites, documenting the task and its requirements, and communicating with reviewers. 
```

> To support users in creating good quality kata and translations, a set of help pages was created with articles related to content authoring activities: tutorials explaining how to create and translate a kata, and reference of [guidelines][guidelines-authoring] and requirements which every author should follow to make sure that a piece of content created by them meets quality criteria.

Reviewer: ```suggestion
To support users in creating good quality kata and translations, a set of help pages were created with articles related to content authoring activities: tutorials explaining how to create and translate a kata, and a reference of [guidelines][guidelines-authoring] and requirements that every author should follow to make sure that a piece of content created by them meets quality criteria.
```

## Creating a Kata

_TBD: improve this stub_

Users new to creating kata can use [tutorial(TODO)](/authoring/kata/).

Some [languages](/languages/) also have a page dedicated to creating a kata, with code examples and best practices.

Every kata author should follow [kata authoring guidelines](/authoring/guidelines/kata/).

> After being published, every kata goes through [beta process](/concepts/kata/beta-process/) and is a subject to a series of [reviews](/curating/kata/).

Reviewer: ```suggestion
After being published, every kata goes through the [beta process](/concepts/kata/beta-process/) and is subject to a series of [reviews](/curating/kata/).
```

# Overview

> All content available on Codewars is created members of Codewars community. It's ideas of Codewars users what gets turned into kata, and only the fact that users are familiar with plenty of programming languages allows for large amount of available translations.

Reviewer: ```suggestion
All content available on Codewars is created by members of the Codewars community. It's ideas of Codewars users what gets turned into kata, and only the fact that users are familiar with plenty of programming languages allows for a large amount of available translations.
```

> To support users in creating good quality of kata and translations, a set of help pages was created with articles related to content authoring activities: tutorials explaining how to create and translate a kata, and reference of guidelines and requirements which every author should follow to make sure that a piece of content created by them meets quality criteria.

Reviewer: ```suggestion
To support users in creating good quality kata and translations, a set of help pages was created with articles related to content authoring activities: tutorials explaining how to create and translate a kata, and reference of guidelines and requirements which every author should follow to make sure that a piece of content created by them meets quality criteria.
```
